packages/xchain-radix/src/client.ts

```diff
@@ -0,0 +1,642 @@
+    } catch (error) {
+      return {
+        from: [],
+        to: [],
+        asset: XrdAsset,
+        date: confirmed_at,
+        type: TxType.Unknown,
+        hash: intent_hash,
+      }
+    }
+  }
+
+  /**
+   * Creates a transaction using the SimpleTransactionBuilder, signs it with the
+   * private key and returns the signed hex
+   * @param params - The transactions params
+   * @returns A signed transaction hex
+   */
+  async transfer(params: TxParams, broadcastTx = true): Promise<string> {
+    const networkId = this.getRadixNetwork()
+    const radixPrivateKey = this.getRadixPrivateKey()
+
+    if (params.asset == undefined) {
+      params.asset = XrdAsset
+    }
+
+    const preparedTransaction = await this.prepareTx(params)
+    const compiledTransactionManifest = await Convert.HexString.toUint8Array(preparedTransaction.rawUnsignedTx)
+    const transactionManifest = await RadixEngineToolkit.TransactionManifest.decompile(
+      compiledTransactionManifest,
+      networkId,
+    )
+    try {
+      const gatewayStatusResponse: GatewayStatusResponse = await this.gatewayApiClient.status.getCurrent()
+      const notarizedTransaction = await TransactionBuilder.new().then((builder) =>
+        builder
+          .header({
+            networkId: networkId,
+            startEpochInclusive: gatewayStatusResponse.ledger_state.epoch,
+            endEpochExclusive: gatewayStatusResponse.ledger_state.epoch + 10,
+            nonce: generateRandomNonce(),
+            notaryPublicKey: radixPrivateKey.publicKey(),
+            notaryIsSignatory: true,
+            tipPercentage: 0,
+          })
+          .plainTextMessage(params.memo ? params.memo : '')
+          .manifest(transactionManifest)
+          .notarize(radixPrivateKey),
+      )
+
+      const compiledTransaction = await RadixEngineToolkit.NotarizedTransaction.compile(notarizedTransaction)
+      const compiledTransactionHex = Convert.Uint8Array.toHexString(compiledTransaction)
+
+      if (broadcastTx) {
+        await this.broadcastTx(compiledTransactionHex)
+      }
+      const transactionId = await RadixEngineToolkit.NotarizedTransaction.intentHash(notarizedTransaction)
+      return transactionId.id
+    } catch (error) {
+      throw new Error('Failed to transfer')
+    }
+  }
+  /**
+   * Submits a transaction
+   * @param txHex - The transaction hex build with the transfer method
+   * @returns - The response from the gateway
+   */
+  async broadcastTx(txHex: string): Promise<string> {
+    await this.gatewayApiClient.transaction.innerClient.transactionSubmit({
+      transactionSubmitRequest: {
+        notarized_transaction_hex: txHex,
+      },
+    })
+    const notarizedTransaction = await RadixEngineToolkit.NotarizedTransaction.decompile(
+      Convert.HexString.toUint8Array(txHex),
+    )
+    const intentHash = await RadixEngineToolkit.NotarizedTransaction.intentHash(notarizedTransaction)
+    return Convert.Uint8Array.toHexString(intentHash.hash)
```

Comment:
Our transaction ids are Bech32m encoded and not hex encoded. You should use `RadixEngineToolkit.Derive.bech32mTransactionIdentifierFromIntentHash` here.

Reply:
I think this is now fixed by your refactor :)